### PR TITLE
Document where to find OSCAL-specific custom metaschema functions not in metaschema java

### LINKED ISSUE #40: Document where to find OSCAL-specific custom Metaschema functions not in metaschema-java
### User Story

As a developer of Metaschema-based tools for OSCAL development, I want to know where I can find a registry of functions and the implementation of each of the functions in the appropriate codebase to understand the algorithm or build alternative implementations.

### Goals

- [ ] Clarify where functions not part of core Metaschema are located in this codebase

### Dependencies

_No response_

### Acceptance Criteria

- [ ] All website and readme documentation affected by the changes in this issue have been updated.
- [ ] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

### Revisions

_No response_

### PR DESCRIPTION
**Pull request by aj-stein-gsa**
# Committer Notes

Closes #40.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/metaschema-framework/liboscal-java/blob/main/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/metaschema-framework/liboscal-java/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you updated all website and readme documentation affected by the changes you made?
- ~Have you written new tests for your core changes, as applicable?~
- ~Have you included examples of how to use your new feature(s)?~

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -71,7 +71,18 @@ Note: One or more pull requests must be submitted addressing all user story goal
...
 ## Communications mechanisms
 
-You can contact the maintainers of this project at [maintainers@metaschema.dev](mailto:maintainers@metaschema.dev) if you are interested in contributing to the development of this project or exchanging ideas. 
+You can contact the maintainers of this project at [maintainers@metaschema.dev](mailto:maintainers@metaschema.dev) if you are interested in contributing to the development of this project or exchanging ideas.
+
+## Developer information
+
+### Core metaschema functions and custom project functions
+
+The Metaschema [specification](https://pages.nist.gov/metaschema/specification/syntax/metapath/) and the conformant [metaschema-java](https://github.com/metaschema-framework/metaschema-java) library implement the Metapath functions required by the specification. However, other libraries, such as this [this project's library](/) can extend the inventory of available functionality with custom functions that are not required in the core specification. These are OSCAL-specific functions, not core Metaschema functions. At the the of the v4.1.0 release, there are only two custom OSCAL-specific functions in this library.
```
Comment (aj-stein-gsa) [inline CONTRIBUTING.md:80]: @david-waltermire, once https://github.com/metaschema-framework/metaschema-java/pull/124 is merged, can we update the URL to where you want it if you expect it into main eventually? I don't want to break link checking, so I won't start out with that.

```suggestion
The Metaschema [specification](https://pages.nist.gov/metaschema/specification/syntax/metapath/) and the conformant [metaschema-java](https://github.com/metaschema-framework/metaschema-java/tree/develop/CONTRIBUTING.md#core-metaschema-functions-functions) library implement the Metapath functions required by the specification. However, other libraries, such as this [this project's library](/) can extend the inventory of available functionality with custom functions that are not required in the core specification. These are OSCAL-specific functions, not core Metaschema functions. At the the of the v4.1.0 release, there are only two custom OSCAL-specific functions in this library.
```

**Comment by RS-Credentive** [comment]
Okay, this clarifies my confusion regarding the definition of "has-oscal-namespace" and profile resolution. These are OSCAL specific metaschema extensions. As an author of metaschema tooling, I would like to see a supported mechanism for defining custom metapath functions that can be automatically processed. I will open a separate issue.

File: CONTRIBUTING.md (hunk 1 above)
Comment (david-waltermire) [reply CONTRIBUTING.md:80]: Note the `[this project's library](/)` still looks not right.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -71,7 +71,18 @@ Note: One or more pull requests must be submitted addressing all user story goal
...
+
+- The [`HasOscalNamespace` class](https://github.com/metaschema-framework/liboscal-java/blob/v4.1.0/src/main/java/gov/nist/secauto/oscal/lib/metapath/function/library/HasOscalNamespace.java) implements the `has-oscal-namespace` function.
+- The [`ResolveProfile` class](https://github.com/metaschema-framework/liboscal-java/blob/v4.1.0/src/main/java/gov/nist/secauto/oscal/lib/metapath/function/library/HasOscalNamespace.java) implements the `resolve-profile()` function.
+
+See the library's [registry class with an inventory of these functions](https://github.com/metaschema-framework/liboscal-java/blob/develop/src/main/java/gov/nist/secauto/oscal/lib/metapath/function/library/OscalFunctionLibrary.java) for a current list of custom, OSCAL-specific functions after the v4.1.0 release.
```
Comment (david-waltermire) [inline CONTRIBUTING.md:85]: Maybe point to `main` here?

File: CONTRIBUTING.md (hunk 1 above)
Comment (aj-stein-gsa) [reply CONTRIBUTING.md:80]: Yeah I spent a few minutes trying different alternatives and none seem to work, I will just add the full URL.

```suggestion
The Metaschema [specification](https://pages.nist.gov/metaschema/specification/syntax/metapath/) and the conformant [metaschema-java](https://github.com/metaschema-framework/metaschema-java) library implement the Metapath functions required by the specification. However, other libraries, such as this [this project's library](https://github.com/metaschema-framework/liboscal-java) can extend the inventory of available functionality with custom functions that are not required in the core specification. These are OSCAL-specific functions, not core Metaschema functions. At the the of the v4.1.0 release, there are only two custom OSCAL-specific functions in this library.
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (aj-stein-gsa) [reply CONTRIBUTING.md:85]:
```suggestion
See the library's [registry class with an inventory of these functions](https://github.com/metaschema-framework/liboscal-java/blob/main/src/main/java/gov/nist/secauto/oscal/lib/metapath/function/library/OscalFunctionLibrary.java) for a current list of custom, OSCAL-specific functions after the v4.1.0 release.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -77,12 +77,12 @@ You can contact the maintainers of this project at [maintainers@metaschema.dev](
...
+The Metaschema [specification](https://pages.nist.gov/metaschema/specification/syntax/metapath/) and the conformant [metaschema-java](https://github.com/metaschema-framework/metaschema-java/tree/develop/CONTRIBUTING.md#core-metaschema-functions-functions) library implement the Metapath functions required by the specification. However, other libraries, such as this [this project's library](https://github.com/metaschema-framework/liboscal-java) can extend the inventory of available functionality with custom functions that are not required in the core specification. These are OSCAL-specific functions, not core Metaschema functions. At the the of the v4.1.0 release, there are only two custom OSCAL-specific functions in this library.
```
Comment (david-waltermire) [inline CONTRIBUTING.md:80]: Should this point to main?
```suggestion
The Metaschema [specification](https://pages.nist.gov/metaschema/specification/syntax/metapath/) and the conformant [metaschema-java](https://github.com/metaschema-framework/metaschema-java/tree/main/CONTRIBUTING.md#core-metaschema-functions-functions) library implement the Metapath functions required by the specification. However, other libraries, such as this [this project's library](https://github.com/metaschema-framework/liboscal-java) can extend the inventory of available functionality with custom functions that are not required in the core specification. These are OSCAL-specific functions, not core Metaschema functions. At the the of the v4.1.0 release, there are only two custom OSCAL-specific functions in this library.
```

Comment (aj-stein-gsa) [reply CONTRIBUTING.md:80]: I am fine either way, but I thought we might want to point this where developers can see what is implemented and not released as well, but I am fine with this.

**Review by david-waltermire — APPROVED** [review]

**Pull request merged**